Customcolorschase

cjsieh opened the pull request:
Allow for a list of colors to be used with the Chase animation.  New animation CustomColorsChase is provided with example.  Issue#50 .

cjsieh committed:
Add files via upload

Add ability to select custom colors for chase animation

cjsieh committed:
Add files via upload

Added CustomColorChase example.  Updated All example with new CustomColorChase

cjsieh committed:
Add files via upload

Changed from customcolorchase to customcolorschase

cjsieh committed:
Delete led_animation_customcolorchase.py

Changed from customcolorchase to customcolorschase

cjsieh committed:
Add files via upload

Changed example CustomColorChase to CustomColorsChase

cjsieh committed:
Add files via upload

Change from CustomColorChase to CustomColorsChase

cjsieh committed:
Delete customcolorchase.py

changed from customcolorchase to customcolorschase

cjsieh committed:
Add files via upload

fixed spaces before/after parameters

cjsieh committed:
Add files via upload

reformatted via black

cjsieh committed:
Add files via upload

reformatted via black

cjsieh committed:
Add files via upload

fixed "line too long" as indicated by pylint

kattni commented on adafruit_led_animation/animation/customcolorschase.py:
```diff
@@ -0,0 +1,85 @@
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_led_animation.animation.rainbowchase`
```
This needs to be updated to match the module name, e.g. `adafruit_led_animation.animation.customcolorchase`.

cjsieh replied:
changed

kattni commented on adafruit_led_animation/animation/customcolorschase.py:
```diff
@@ -0,0 +1,85 @@
+`adafruit_led_animation.animation.rainbowchase`
+================================================================================
+
+Rainbow chase animation for CircuitPython helper library for LED animations.
```
This needs to be updated to describe the module, e.g. `Custom color chase animation for CircuitPython helper library for LED animations.`.

cjsieh replied:
changed

kattni commented on adafruit_led_animation/animation/customcolorschase.py:
```diff
@@ -0,0 +1,85 @@
+from adafruit_led_animation.color import RAINBOW
+
+
+class CustomColorsChase(Chase):
```
Class names are typically singular. Please update this to `CustomColorChase`. This will also require updating all subsequent use of the class name.

cjsieh replied:
changed

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+# Patriotic
+custom_colors_chase_rwb = CustomColorsChase(
+    pixels, speed=0.1, colors=[RED, WHITE, BLUE], size=2, spacing=3
+)
```
We have a worldwide community and want everyone to feel welcome and included. Including this color combination under this name limits the scope of the example to the US, and could create a situation where users from elsewhere in the world, or even in the US, feel excluded. In light of the current climate, I feel it should be removed altogether.

cjsieh replied:
changed

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+custom_colors_chase_rwb = CustomColorsChase(
+    pixels, speed=0.1, colors=[RED, WHITE, BLUE], size=2, spacing=3
+)
+# St Pat Day
```
Associating these color combinations with holidays could be exclusionary as some do not celebrate them. The colors are fine, but please remove the holiday themed comments.

cjsieh replied:
changed

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+custom_colors_chase_gw = CustomColorsChase(
+    pixels, speed=0.1, colors=[GREEN, WHITE], size=2, spacing=3
+)
+# Christmas
```
Same as above, re: removing holiday themed comments.

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+custom_colors_chase_rg_r = CustomColorsChase(
+    pixels, speed=0.1, colors=[RED, GREEN], size=1, spacing=2, reverse=True
+)
+# Valentines Day
```
Same as above, re: removing holiday themed comments.

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+custom_colors_chase_rg = CustomColorsChase(
+    pixels, speed=0.1, colors=[RED, GREEN], size=2, spacing=3
+)
+custom_colors_chase_rg_r = CustomColorsChase(
```
Consider renaming this to `custom_colors_chase_rg_reversed`. It's long, but the difference between the two is clearer.

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+)
+
+# colors default to RAINBOW
+custom_colors_chase_rainbow = CustomColorsChase(pixels, speed=0.1, size=2, spacing=3)
```
I would update this to `custom_color_chase_rainbow` - removing the `s` from `colors` to match the class rename. Same suggestion goes for all instances of `custom_colors` in this example.

kattni commented on examples/led_animation_all_animations.py:
```diff
@@ -43,6 +44,9 @@
 rainbow_comet = RainbowComet(pixels, speed=0.1, tail_length=7, bounce=True)
 rainbow_chase = RainbowChase(pixels, speed=0.1, size=3, spacing=2, step=8)
 rainbow_sparkle = RainbowSparkle(pixels, speed=0.1, num_sparkles=15)
+custom_colors_chase = CustomColorsChase(
```
Call this `custom_color_chase` to match the class rename. (As well, the class name will need to be updated.)

cjsieh replied:
changed

kattni commented on examples/led_animation_all_animations.py:
```diff
@@ -57,6 +61,7 @@
     rainbow_comet,
     sparkle_pulse,
     rainbow_chase,
+    custom_colors_chase,
```
Update this to match, e.g. `custom_color_chase`.

cjsieh replied:
changed

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+"""
+This example displays the basic animations in sequence, at a five second interval.
```
This comment needs to be updated to describe the new example.

cjsieh replied:
fixed

kattni commented on examples/led_animation_customcolorschase.py:
```diff
@@ -0,0 +1,61 @@
+
+from adafruit_led_animation.animation.customcolorschase import CustomColorsChase
+from adafruit_led_animation.sequence import AnimationSequence
+from adafruit_led_animation.color import PINK, PURPLE, GREEN, RED, WHITE, BLUE
```
`PURPLE` is not used. If you do not intend to use it, please remove it from the imports.

cjsieh replied:
removed

kattni commented:
I am working on reviewing your code. The current Pylint failure is not in your code. I am submitting a PR with the fix.

FoamyGuy reviewed (approved):
I tested these successfully with a Neopixel  Featherwing, and with a 30 pixel strip. Really neat animation, thank you for adding it @cjsieh!

kattni committed:
Add sequence fix.

kattni reviewed (changes requested):
Hello, @cjsieh! Thanks so much for contributing this animation. I've tested it and it works successfully. There are a few changes I'd like to see before we merge it. Please see the suggestions below. **Note:** I committed a fix to a separate file that was causing Pylint to fail on a file you had not edited. The new Pylint failure is addressed in the comments below. Make sure you update your local version `sequence.py` to match the new version in the PR.

With the new example, I think it would be better to choose some other nice color combinations and include them without themed names.

Let me know if you need any assistance with any of the changes. I'm happy to  help!

cjsieh committed:
Add files via upload

remove references to rainbowchase.  Change customcolorschase to customcolorchase.

cjsieh committed:
Delete customcolorchase.py

put in wrong place

cjsieh committed:
Add files via upload

Changed references to rainbowchase.  Changed from CustomColorsChase to CustomColorChase

cjsieh committed:
Delete customcolorschase.py

changed from customcolorschase to customcolorchase

cjsieh committed:
Add files via upload

change from CustomColorsChase to CustomColorChase

cjsieh committed:
Delete led_animation_customcolorschase.py

changed from ccustomcolorschase to customcolorchase

cjsieh committed:
Add files via upload

prefixed "black" formating

cjsieh committed:
Add files via upload

fixed description
removed reference to advance_on_cycle_complete as not used now

kattni reviewed (approved):
Thank you for the updates!
